src/plugins/slack/fetch.js

```diff
@@ -0,0 +1,168 @@
+// @flow 
+
+const { WebClient } = require('@slack/web-api')
+import type { 
+  SlackToken, 
+  User, 
+  Conversation, 
+  Message
+} from "./models"
+
+export class Fetcher {
+  +_token: SlackToken;
+  web: any
+
+  constructor(token: SlackToken) {
+    this._token = token
+    if (!token) {
+      throw new Error ("A Slack bot token is required");
+    }
+    this.web = new WebClient(token)
+  }
+
+  /**
+   * Get all users from the Slack organisation
+   */
+  async users(): Promise<Array<User>> {
+    const users: any = [];
+    let cursor: any;
+    let response: any;
+    do {
+      response = await this.web.users.list({
+        limit: 100,
+        include_locale: false,
+        cursor
+      });
+      const onlyHumans = await response.members.filter((member) => {
+        return member.is_bot !== true && member.profile.real_name_normalized !== 'Slackbot';
+      })
+      for (const human of onlyHumans) {
+        const oneHuman = {
+          id: human.id, 
+          name: human.profile.real_name_normalized,
+          email: human.profile.email 
+        };
+        users.push(oneHuman);
+      }
+      cursor = response.response_metadata.next_cursor;
+    } while (response.response_metadata.next_cursor);
+    return users;
+  }
```

Comment:
You might have mentioned to me that slack auth tokens don't automatically have access to all channels in an org. Might be good to explain the semantics of that here if there is extra config needed to grant access to individual channels, public or private.